als-actions/shared/src/main/scala/org/mulesoft/als/actions/hover/PatchedHover.scala

```diff
@@ -47,4 +55,34 @@ class PatchedHover(provider: VocabularyProvider) {
       AsyncApi20Dialect.dialect.id  -> "asyncapi2"
     )
 
+  def indexAliasTerms(valueType: ValueType,
+                      aliases: Option[Aliases],
+                      dialectName: String,
+                      metaName: String): VocabularyProvider = {
+    if (aliases.nonEmpty) {
+      provider match {
+        case v: AlsVocabularyRegistry => {
+          val hoverProvider = v.branch
+          val generatedTerms: Seq[ClassTerm] = aliases
+            .map(list => {
+              list.aliases
+                .map(alias => {
+                  val aliasTerm = buildAliasTerm(alias._1, dialectName, metaName)
+                  ClassTerm()
+                    .withName(aliasTerm)
+                    .withDisplayName(aliasTerm)
+                    .withDescription(Raml10UsesKeyTerm.description)
+                })
+                .toSeq
+            })
+            .getOrElse(Seq.empty)
+          hoverProvider.index(valueType.ns.base, generatedTerms)
+          hoverProvider
+        }
+        case _ => provider
+      }
+    } else provider
+  }
+
+  def buildAliasTerm(name: String, dialectName: String, metaName: String): String = s"$dialectName/${metaName}/$name"
```

Comment:
ask if there is already such a method in AMF just in case (so we don't have redundant logic)

Reply:
The logic of having URIs to patched terms be `{base}/#{dialect}/{objectMetaName}/{currentAstKey}` was created by us, don't think AMF would have a method to do this.

Reply:
my bad!
are we just using this here? why is this new if we were already doing it?

Reply:
Yes, you are right. We could probably reuse some of the logic here:
https://github.com/aml-org/als/blob/66f516f6f0b569a3817a67131088b019169491ce/als-actions/shared/src/main/scala/org/mulesoft/als/actions/hover/PatchedHover.scala#L46-L47